src/net_processing.cpp

```diff
@@ -892,7 +893,7 @@ class PeerManagerImpl final : public PeerManager
      */
     bool BlockRequested(NodeId nodeid, const CBlockIndex& block, std::list<QueuedBlock>::iterator** pit = nullptr) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
-    bool TipMayBeStale() EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+    bool TipMayBeStale(bool& super_stale) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Comment:
I think `super_stale` is not very descriptive. I also think this pattern of returning extra data from this function in this way is a little confusing. This could be made more descriptive by declaring a new enum and using it as the return value. eg
```
enum Staleness {
    NotStale,
    Stale,
    StalePastNetworkLimited,
};
```

Reply:
yeah, better. Done as suggested